components/director/internal/domain/onetimetoken/service.go

```diff
@@ -48,6 +84,20 @@ func (s service) GenerateOneTimeToken(ctx context.Context, id string, tokenType
 		return model.OneTimeToken{}, errors.Wrap(err, "while creating System Auth")
 	}
 
+	if tokenType == model.ApplicationReference {
+		app, err := s.appSvc.Get(ctx, id)
+
+		if err != nil {
+			return model.OneTimeToken{}, errors.Wrapf(err, "while getting application [id: %s]", id)
+		}
+
+		if app.IntegrationSystemID != nil {
+			if adapterURL, ex := s.intSystemToAdapterMapping[*app.IntegrationSystemID]; ex {
```

Comment:
Maybe we should stick to the pattern with `ok`. To be honest `ex` for me points more to the exceptions.
```suggestion
			if adapterURL, ok := s.intSystemToAdapterMapping[*app.IntegrationSystemID]; ok {
```